geomesa-accumulo/geomesa-accumulo-compute/src/main/scala/org/locationtech/geomesa/compute/spark/GeoMesaSpark.scala

```diff
@@ -123,13 +123,19 @@ object GeoMesaSpark extends LazyLogging {
           InputConfigurator.setAutoAdjustRanges(classOf[AccumuloInputFormat], conf, false)
           InputConfigurator.setAutoAdjustRanges(classOf[GeoMesaAccumuloInputFormat], conf, false)
         }
+
+        org.apache.accumulo.core.client.mapreduce.lib.impl.InputConfigurator.setBatchScan(classOf[AccumuloInputFormat], conf, true)
+        org.apache.accumulo.core.client.mapreduce.lib.impl.InputConfigurator.setBatchScan(classOf[GeoMesaAccumuloInputFormat], conf, true)
         GeoMesaConfigurator.setSerialization(conf)
         GeoMesaConfigurator.setTable(conf, qp.table)
         GeoMesaConfigurator.setDataStoreInParams(conf, dsParams)
         GeoMesaConfigurator.setFeatureType(conf, sft.getTypeName)
-        if (query.getFilter != Filter.INCLUDE) {
-          GeoMesaConfigurator.setFilter(conf, ECQL.toCQL(query.getFilter))
-        }
+
+        // set the secondary filter if it exists and is  not Filter.INCLUDE
+        qp.filter.secondary
+          .collect { case f if f != Filter.INCLUDE => f }
```

Comment:
doesn't really hurt, but `Filter.INCLUDE` should never occur in either filter.primary or filter.secondary - instead there would be a None